.github/workflows/ubuntu-unit-tests.sh

```diff
@@ -0,0 +1,87 @@
+#!/bin/bash
+
+PHASES=(${@:-SETUP RUN RUN_ASAN_UBSAN CLEANUP})
+RELEASE="$(lsb_release -cs)"
+ADDITIONAL_DEPS=(
+    clang
+    expect
+    fdisk
+    libfdisk-dev
+    libfido2-dev
+    libp11-kit-dev
+    libpwquality-dev
+    libqrencode-dev
+    libssl-dev
+    libtss2-dev
+    libzstd-dev
+    perl
+    python3-libevdev
+    python3-pyparsing
+    zstd
+)
+
+function info() {
+    echo -e "\033[33;1m$1\033[0m"
+}
+
+set -ex
+
+for phase in "${PHASES[@]}"; do
+    case $phase in
+        SETUP)
+            info "Setup phase"
+            bash -c "echo 'deb-src http://archive.ubuntu.com/ubuntu/ $RELEASE main restricted universe multiverse' >>/etc/apt/sources.list"
+            # PPA with some newer build dependencies
+            add-apt-repository -y ppa:upstream-systemd-ci/systemd-ci
+            apt-get -y update
+            apt-get -y build-dep systemd
+            apt-get -y install "${ADDITIONAL_DEPS[@]}"
+
+            # test_exec_dynamicuser (test-execute) in GH Actions suffers from
+            # the same issue as described in systemd/systemd#10337. Let's
+            # temporarily skip this particular subtest until a nicer solution
+            # is implemented.
+            # For the record, at the time of writing the `find` in the
+            # exec-dynamicuser-statedir.service runs for almost 6 minutes.
+            sed -i '/void test_exec_dynamicuser/a\        return;' src/test/test-execute.c
```

Comment:
can you change this too to do a skip via env var, please?

Reply:
ah, yes, completely forgot about this one, will do